code/modules/reagents/reagents/reagents_basic.dm

```diff
@@ -338,8 +338,6 @@
 	else if(istype(O, /obj/item/weapon/book/manual/snow))
 		var/obj/item/weapon/book/manual/snow/S = O
 		S.trigger()
-	else if(O.on_fire) // For extinguishing objects on fire
```

Comment:
Please double check that this doesn't break foam fire extinguishers

Reply:
that change in particular won't break them since clean_act() up above calls extinguish() now.

Reply:
![image](https://github.com/vgstation-coders/vgstation13/assets/7573912/b960f152-2cc9-4660-bf81-3618eb0b433b)
yeah that works